documentation/openapi-authoring-automated-guidelines.md

```diff
@@ -261,4 +263,302 @@ Links: [Index](#index) | [Error vs. Warning](#error-vs-warning) | [Automated Rul
+
+**Impact on generated code**: N/A.
+
+**Examples**: N/A.
+
+Links: [Index](#index) | [Error vs. Warning](#error-vs-warning) | [Automated Rules](#automated-rules) | [RPC](#rpc-violations): [Errors](#rpc-errors) or [Warnings](#rpc-warnings) | [SDK](#sdk-violations): [Errors](#sdk-errors) or [Warnings](#sdk-warnings)
+
+### <a name="R1009" />R1009 DeleteInOperationName
+**Output Message**: 'DELETE' operation '{0}' should use method name 'Delete'. Note: If you have already shipped an SDK on top of this spec, fixing this warning may introduce a breaking change.
+
+**Description**: Verifies whether value for `operationId` is named as per ARM guidelines.
+
+**Why the rule is important**: Per ARM guidelines, each 'DELETE' operation on a resource should have "delete" in the name.
+
+**How to fix the violation**: Make sure that `operationId` is in the form of `NOUN_Delete` or `Delete`.
+
+**Impact on generated code**: Operation name in the generated SDK will be named based on this.
+
+**Examples**:
+* Resources_Delete
+* Delete
+* StorageAccounts_delete
+* delete
+
+Links: [Index](#index) | [Error vs. Warning](#error-vs-warning) | [Automated Rules](#automated-rules) | [RPC](#rpc-violations): [Errors](#rpc-errors) or [Warnings](#rpc-warnings) | [SDK](#sdk-violations): [Errors](#sdk-errors) or [Warnings](#sdk-warnings)
+
+### <a name="R1005" />R1005 GetInOperationName
+**Output Message**: 'GET' operation '{0}' should use method name 'Get' or Method name start with 'List'. Note: If you have already shipped an SDK on top of this spec, fixing this warning may introduce a breaking change.
+
+**Description**: Verifies whether value for `operationId` is named as per ARM guidelines.
+
+**Why the rule is important**: Per ARM guidelines, each 'GET' operation on a resource should have "get" or "list" in the name.
+
+**How to fix the violation**: Make sure that `operationId` is in the form of `NOUN_Get`, `NOUN_List`, `Get` or `List`.
+
+**Impact on generated code**: Operation name in the generated SDK will be named based on this.
+
+**Examples**:
+* Resources_Get
+* Resources_List
+* Get
+* List
+
+Links: [Index](#index) | [Error vs. Warning](#error-vs-warning) | [Automated Rules](#automated-rules) | [RPC](#rpc-violations): [Errors](#rpc-errors) or [Warnings](#rpc-warnings) | [SDK](#sdk-violations): [Errors](#sdk-errors) or [Warnings](#sdk-warnings)
+
+### <a name="R1003" />R1003 ListInOperationName
+**Output Message**: Since operation '{0}' response has model definition '{1}', it should be of the form "*_list*".
+
+**Description**: Verifies whether value for `operationId` is named as per ARM guidelines when response contains array of items.
+
+**Why the rule is important**: Per ARM guidelines, each 'GET' operation on a resource should have "list" in the name when operation has [x-ms-pageable](https://github.com/Azure/autorest/tree/master/docs/extensions#x-ms-pageable) extension.
```

Comment:
similar to above, when we refer to ARM guidelines we generally refer to the REST APIs and not SDKs, in this case, these are SDK guidelines, so I would clarify it here. And in terms of why, we're aiming for consistency. 
Since all these naming rules are warnings, just double check any strong wording.